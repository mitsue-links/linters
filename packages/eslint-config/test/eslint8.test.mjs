import path from 'path';
import {lintFile} from './helper.mjs';

describe('ESLint 8.0.0', () => {
    test('8.0.0_ok.mjs', async () => {
        const filePath = '8.0.0_ok.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('8.0.0_error.mjs', async () => {
        const filePath = '8.0.0_error.mjs';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 8.1.0', () => {
    test('8.1.0_ok.js', async () => {
        const filePath = '8.1.0_ok.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('8.1.0_error.js', async () => {
        const filePath = '8.1.0_error.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});

describe('ESLint 8.5.0', () => {
    test('8.5.0_ok.js', async () => {
        const filePath = '8.5.0_ok.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toStrictEqual([{
            filePath,
            messages: [],
        }]);
    });

    test('8.5.0_error.js', async () => {
        const filePath = '8.5.0_error.js';
        const data = await lintFile(path.join('fixtures', 'eslint', filePath));

        expect(data).toMatchSnapshot();
    });
});
