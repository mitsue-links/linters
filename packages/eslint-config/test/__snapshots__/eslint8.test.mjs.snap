// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint 8.0.0 8.0.0_error.mjs 1`] = `
[
  {
    "filePath": "8.0.0_error.mjs",
    "messages": [
      {
        "line": 3,
        "ruleId": "class-methods-use-this",
        "severity": 1,
      },
      {
        "line": 7,
        "ruleId": "class-methods-use-this",
        "severity": 1,
      },
    ],
  },
]
`;

exports[`ESLint 8.1.0 8.1.0_error.js 1`] = `
[
  {
    "filePath": "8.1.0_error.js",
    "messages": [
      {
        "line": 6,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      {
        "line": 8,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      {
        "line": 12,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
      {
        "line": 14,
        "ruleId": "no-unused-private-class-members",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`ESLint 8.5.0 8.5.0_error.js 1`] = `
[
  {
    "filePath": "8.5.0_error.js",
    "messages": [
      {
        "line": 9,
        "ruleId": "prefer-object-has-own",
        "severity": 2,
      },
    ],
  },
]
`;
